src/v/archival/ntp_archiver_service.cc

```diff
@@ -1449,76 +1450,16 @@ static ss::future<ntp_archiver_upload_result> aggregate_upload_results(
 }
 
 ss::future<ntp_archiver::scheduled_upload>
-ntp_archiver::schedule_single_upload(const upload_context& upload_ctx) {
-    auto start_upload_offset = upload_ctx.start_offset;
-    auto last_stable_offset = upload_ctx.last_offset;
-
-    auto log = _parent.log();
-
-    upload_candidate_with_locks upload_with_locks;
-    switch (upload_ctx.upload_kind) {
-    case segment_upload_kind::non_compacted:
-        upload_with_locks = co_await _policy.get_next_candidate(
-          start_upload_offset,
-          last_stable_offset,
-          log,
-          *_parent.get_offset_translator_state(),
-          _conf->segment_upload_timeout);
-        break;
-    case segment_upload_kind::compacted:
-        const auto& m = manifest();
-        upload_with_locks = co_await _policy.get_next_compacted_segment(
-          start_upload_offset, log, m, _conf->segment_upload_timeout);
-        break;
-    }
-
-    auto upload = upload_with_locks.candidate;
-    auto locks = std::move(upload_with_locks.read_locks);
-
-    if (
-      upload.sources.empty()
-      && upload_ctx.upload_kind == segment_upload_kind::compacted
-      && model::offset{} != upload.final_offset) {
-        vlog(
-          _rtclog.warn,
-          "Upload skipped for range: {}-{} because these offsets lie inside "
-          "batches or the reupload is not smaller than the current segment",
-          upload.starting_offset,
-          upload.final_offset);
-        co_return scheduled_upload{
-          .result = std::nullopt,
-          .inclusive_last_offset = upload.final_offset,
-          .meta = std::nullopt,
-          .name = std::nullopt,
-          .delta = std::nullopt,
-          .stop = ss::stop_iteration::yes,
-          .upload_kind = upload_ctx.upload_kind,
-        };
-    }
-
-    if (upload.sources.empty()) {
-        vlog(
-          _rtclog.debug,
-          "upload candidate not found, start_upload_offset: {}, "
-          "last_stable_offset: {}",
-          start_upload_offset,
-          last_stable_offset);
-        // Indicate that the upload is not started
-        co_return scheduled_upload{
-          .result = std::nullopt,
-          .inclusive_last_offset = {},
-          .meta = std::nullopt,
-          .name = std::nullopt,
-          .delta = std::nullopt,
-          .stop = ss::stop_iteration::yes,
-          .upload_kind = upload_ctx.upload_kind,
-        };
-    }
+ntp_archiver::maybe_schedule_single_upload(
```

Comment:
not sure i understand why this is named `maybe_schedule_single_upload` after we removed the other overloads

what do you think about naming it `(do_)upload_candidate`?

Reply:
yeah, it should be renamed now that overloads are not present, fixed